cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -361,8 +373,25 @@ func (c *CStorVolumeReplicaController) cVRAddEventHandler(
 			string(common.AlreadyPresent),
 			string(common.MessageResourceAlreadyPresent),
 		)
-		// If the volume already present then return the cvr status as online
-		return string(apis.CVRStatusOnline), nil
+		// After creating zfs datasets in zpool but update to etcd might be
+		// failed
+		if isEmptyReplicaID(cVR) {
```

Comment:
Can you move this code to separate function? This add handler function has become lengthy.

Reply:
Yes, I also worried about the same thing while touching this function. We will take care while refactoring the entire CVR reconciliation code.
```
//TODO: Follow best practice while refactoring reconciliation logic
```
Added above line the same function.

Reply:
done